cmd/podman/trust.go

```diff
@@ -0,0 +1,293 @@
+		}
+		return nil
+	}
+
+	var policyContentStruct trust.PolicyContent
+	if err := json.Unmarshal(policyContent, &policyContentStruct); err != nil {
+		return errors.Errorf("could not read trust policies")
+	}
+	policyJSON, err := trust.GetPolicyJSON(policyContentStruct, systemRegistriesDirPath)
+	if err != nil {
+		return errors.Wrapf(err, "error reading registry config file")
+	}
+	if c.Bool("json") {
+		var outjson interface{}
+		outjson = policyJSON
+		out := formats.JSONStruct{Output: outjson}
+		return formats.Writer(out).Out()
+	}
+
+	sortedRepos := sortPolicyJSONKey(policyJSON)
+	type policydefault struct {
+		Repo      string
+		Trusttype string
+		GPGid     string
+		Sigstore  string
+	}
+	var policyoutput []policydefault
+	for _, repo := range sortedRepos {
+		repoval := policyJSON[repo]
+		var defaultstruct policydefault
+		defaultstruct.Repo = repo
+		if repoval["type"] != nil {
+			defaultstruct.Trusttype = trustTypeDescription(repoval["type"].(string))
+		}
+		if repoval["keys"] != nil && len(repoval["keys"].([]string)) > 0 {
+			defaultstruct.GPGid = trust.GetGPGId(repoval["keys"].([]string))
+		}
+		if repoval["sigstore"] != nil {
+			defaultstruct.Sigstore = repoval["sigstore"].(string)
+		}
+		policyoutput = append(policyoutput, defaultstruct)
+	}
+	var output []interface{}
+	for _, ele := range policyoutput {
+		output = append(output, interface{}(ele))
+	}
+	out := formats.StdoutTemplateArray{Output: output, Template: "{{.Repo}}\t{{.Trusttype}}\t{{.GPGid}}\t{{.Sigstore}}"}
+	return formats.Writer(out).Out()
+}
+
+func setTrustCmd(c *cli.Context) error {
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not create runtime")
+	}
+
+	args := c.Args()
+	if len(args) != 1 {
+		return errors.Errorf("default or a registry name must be specified")
+	}
+	valid, err := image.IsValidImageURI(args[0])
+	if err != nil || !valid {
+		return errors.Wrapf(err, "invalid image uri %s", args[0])
+	}
+
+	trusttype := c.String("type")
+	if !isValidTrustType(trusttype) {
+		return errors.Errorf("invalid choice: %s (choose from 'accept', 'reject', 'signedBy')", trusttype)
+	}
+	if trusttype == "accept" {
+		trusttype = "insecureAcceptAnything"
+	}
+
+	pubkeysfile := c.StringSlice("pubkeysfile")
+	if len(pubkeysfile) == 0 && trusttype == "signedBy" {
+		return errors.Errorf("At least one public key must be defined for type 'signedBy'")
+	}
+
+	var policyPath string
+	if c.IsSet("policypath") {
+		policyPath = c.String("policypath")
+	} else {
+		policyPath = trust.DefaultPolicyPath(runtime.SystemContext())
+	}
+	var policyContentStruct trust.PolicyContent
+	_, err = os.Stat(policyPath)
+	if !os.IsNotExist(err) {
+		policyContent, err := ioutil.ReadFile(policyPath)
+		if err != nil {
+			return errors.Wrapf(err, "unable to read %s", policyPath)
+		}
+		if err := json.Unmarshal(policyContent, &policyContentStruct); err != nil {
+			return errors.Errorf("could not read trust policies")
+		}
+	}
```

Comment:
Does this leave `policyContentStruct.Default` empty when the policy does not exist? That is invalid and it would cause all signature users to reject to process any signatures.

(Granted, it’s unlikely that the file will be missing; but if it is, this code should probably force the user to set the default before accepting any other operations.)